pkg/operator/ceph/cluster/osd/remove.go

```diff
@@ -156,27 +156,27 @@ func waitForRebalance(context *clusterd.Context, namespace string, osdID int, in
 	// wait until the cluster gets fully rebalanced again
 	err := util.Retry(3000, 15*time.Second, func() error {
```

Comment:
What is the reason for having this retry as 3000?

Reply:
The cluster might take a while to rebalance, so I guess this is just a long value.

Reply:
But this will stop other operations(e.g., adding a node or device) to take place while the cluster is rebalancing.

Reply:
Yeah, operations like this that could take a really long time should be asynchronous. As a separate issue, we should look into make this asynchronous. At the same time we really need to evaluate when we even want rook to automatically remove OSDs.